documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -398,6 +401,114 @@ long executionId = BatchRuntime
 ----
 ====
 
+==== Job Parameters
+
+.Job Parameters in JSR 352 Integration
+|===
+|Parameter Name |Builder Method |Requirement |Default value |Description
+
+|`rootEntities`
+|`forEntity(Class<?>)`, `forEntities(Class<?>, Class<?>...)`
+|Required
+|-
+|The root entity types to index in this job execution.
+
+|`purgeAllOnStart`
+|`purgeAllOnStart(boolean)`
+|Optional
+|False
+|Specify whether the existing lucene index should be purged at the beginning of the job. This
+operation takes place before the step of lucene document production.
+
+|`optimizeAfterPurge`
+|`optimizeAfterPurge(boolean)`
+|Optional
+|False
+|Specify whether the mass indexer should be optimized at the beginning of the job. This operation
+takes place after the purge operation and before the step of lucene document production.
+
+|`optimizeOnFinish`
+|`optimizeOnFinish(boolean)`
+|Optional
+|False
+|Specify whether the mass indexer should be optimized at the end of the job. This operation takes
+place after the step of lucene document production.
+
+|`cacheable`
+|`cacheable(boolean)`
+|Optional
+|False
+|Specify whether the Hibernate queries are cacheable. This setting will be applied to the internal
+entity reader class. Set it to true when reading a complex graph with relations.
+
+|`fetchSize`
+|`fetchSize(int)`
+|Optional
+|200,000
+|The size for the result fetching.
+
+|`customQueryHQL`
+|`restrictedBy(String)`
+|Optional
+|-
+|Use HQL / JPQL to select the entities to index. Mixing this approach with the criteria restriction
+is not allowed. Please notice that there's no query validation for your input.
+
+|`customQueryCriteria`
+|`restrictedBy(Criterion)`
+|Optional
+|-
+|Add criterion to construct a customized selection of mass-indexing under the criteria approach. You
+can call this method as many times as the number of criterion wished. However, mixing this approach
+with the HQL restriction is not allowed.
+
+|`customQueryLimit`
+|`customQueryLimit(int)`
+|Optional
+|1,000,000
+|The limit of number of entities read in a custom query. This parameter is only used under
+custom query approaches (HQL and criteria). The value defined must be greater than 0.
+
+|`checkpointInterval`
+|`checkpointInterval(int)`
+|Optional
+|200
+|The number of entities read before trigger a checkpoint. The value defined must be greater than 0,
```

Comment:
I just checked `org.hibernate.search.jsr352.massindexing.impl.steps.lucene.CheckpointAlgorithm` to be safe, and I was a bit confused at first because it uses `MetricType.READ_COUNT`.
But actually, it seems items are only ever written upon checkpoints. Between checkpoints, items are only read and processed, the output being stored in a list, waiting for the next checkpoint, when it will be written.
Which is why you can only use the "read count" metric to determine whether to perform a checkpoint or not: it is the only metric that will grow between checkpoints

So... It seems "read" was the right word actually (though "processed" would technically be right too), but maybe we should refine the definition of what is a "checkpoint" (further down the page). It seems like an important fact to me: items are only written when a checkpoint occurs (be it at the end of a partition or at the end of a checkpoint interval). This means in particular that disabling partitioning (very high "rows per partition") and also disabling checkpoints (very high "checkpointInterval") will likely take *lots* of memory.